src/handler/v2/edition_test.go

```diff
@@ -0,0 +1,1927 @@
+				assert.Equal(t, testCase.expectEdition.Questionnaire, res.Questionnaire)
+				assert.WithinDuration(t, testCase.expectEdition.CreatedAt, res.CreatedAt, time.Second)
+			}
+		})
+	}
+}
+
+func TestDeleteEdition(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockEditionService := mock.NewMockEdition(ctrl)
+	edition := NewEdition(mockEditionService)
+
+	editionID := uuid.New()
+
+	type test struct {
+		description       string
+		editionID         openapi.EditionIDInPath
+		executeDeleteMock bool
+		launcherVersionID values.LauncherVersionID
+		deleteEditionErr  error
+		isErr             bool
+		statusCode        int
+	}
+
+	testCases := []test{
+		{
+			description:       "特に問題ないのでエラー無し",
+			editionID:         editionID,
+			executeDeleteMock: true,
+			launcherVersionID: values.NewLauncherVersionIDFromUUID(editionID),
+			statusCode:        http.StatusOK,
+		},
+		{
+			description:       "存在しないエディションIDなので400",
+			editionID:         editionID,
+			executeDeleteMock: true,
+			launcherVersionID: values.NewLauncherVersionIDFromUUID(editionID),
+			deleteEditionErr:  service.ErrInvalidEditionID,
+			isErr:             true,
+			statusCode:        http.StatusBadRequest,
+		},
+		{
+			description:       "DeleteEditionがエラーなので500",
+			editionID:         editionID,
+			executeDeleteMock: true,
+			launcherVersionID: values.NewLauncherVersionIDFromUUID(editionID),
+			deleteEditionErr:  errors.New("internal error"),
+			isErr:             true,
+			statusCode:        http.StatusInternalServerError,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			if testCase.executeDeleteMock {
+				mockEditionService.
+					EXPECT().
+					DeleteEdition(gomock.Any(), testCase.launcherVersionID).
+					Return(testCase.deleteEditionErr)
+			}
+
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodDelete, fmt.Sprintf("/api/v2/editions/%s", testCase.editionID), nil)
+			rec := httptest.NewRecorder()
+			c := e.NewContext(req, rec)
+
+			err := edition.DeleteEdition(c, testCase.editionID)
+
+			if testCase.isErr {
+				if testCase.statusCode != 0 {
+					var httpErr *echo.HTTPError
+					if errors.As(err, &httpErr) {
+						assert.Equal(t, testCase.statusCode, httpErr.Code)
+					} else {
+						t.Errorf("error should be *echo.HTTPError, but got %T", err)
+					}
+				} else {
+					assert.Error(t, err)
+				}
+				return
+			}
+
+			assert.NoError(t, err)
+			assert.Equal(t, http.StatusOK, rec.Code)
+		})
+	}
+}
+
+func TestGetEdition(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockEditionService := mock.NewMockEdition(ctrl)
+	edition := NewEdition(mockEditionService)
+
+	type test struct {
+		description   string
+		editionID     openapi.EditionIDInPath
+		resultEdition *domain.LauncherVersion
+		getEditionErr error
+		expectEdition *openapi.Edition
```

Comment:
実際のメソッド名に合わせて`GetEditionErr`の方がいいかもです
```suggestion
		GetEditionErr error
```